eagle-core/eagle-app/eagle-app-base/src/main/java/org/apache/eagle/app/resource/ApplicationResource.java

```diff
@@ -18,24 +18,55 @@
 
 
 import com.google.inject.Inject;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
 import org.apache.eagle.app.service.ApplicationManagementService;
 import org.apache.eagle.app.service.ApplicationOperations;
 import org.apache.eagle.app.service.ApplicationProviderService;
 import org.apache.eagle.metadata.model.ApplicationDesc;
 import org.apache.eagle.metadata.model.ApplicationEntity;
 import org.apache.eagle.metadata.resource.RESTResponse;
 import org.apache.eagle.metadata.service.ApplicationEntityService;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import javax.ws.rs.*;
 import javax.ws.rs.core.MediaType;
 import java.util.Collection;
 
+import static org.apache.eagle.metadata.utils.HttpRequest.httpGetWithoutCredentials;
+
 @Path("/apps")
 public class ApplicationResource {
+    private final static Logger LOG = LoggerFactory.getLogger(ApplicationResource.class);
+
     private final ApplicationProviderService providerService;
     private final ApplicationManagementService applicationManagementService;
     private final ApplicationEntityService entityService;
 
+    private ApplicationEntity.Status extractStatus(JSONObject response, String appUuid) {
```

Comment:
No logic code should exist in web resource class (`Controller`), push down business logic into `Service` .